ext/ddtrace_profiling_native_extension/heap_recorder.c

```diff
@@ -141,9 +145,11 @@ static int st_object_record_entry_free(st_data_t, st_data_t, st_data_t);
 static int st_object_record_entry_free_if_invalid(st_data_t, st_data_t, st_data_t);
 static int st_object_records_iterate(st_data_t, st_data_t, st_data_t);
 static int update_object_record_entry(st_data_t*, st_data_t*, st_data_t, int);
-static void commit_allocation(heap_recorder*, heap_record*, long, live_object_data);
+static void commit_allocation(heap_recorder*, heap_record*, long, live_object_data*);
 static void flush_queue(heap_recorder*);
-static bool enqueue_allocation(heap_recorder*, heap_record*, long, live_object_data);
+static bool enqueue_allocation(heap_recorder*, heap_record*, long, live_object_data*);
```

Comment:
Minor: I wonder if `commit_allocation` and similar methods should receive an `object_record*` directly?

That way we would only need to pass 1 argument around, instead of having to keep "slicing" the object record's fields to pass them around until we actually create the object record :)

(I'm reminded of the horse scene from "the cell" which always made me :cold_sweat: )

Reply:
Yeah this makes sense and allows me to remove yet another struct!